devops/jobs/DockerCI.groovy

```diff
@@ -0,0 +1,99 @@
+/*
+    Variables consumed from the EXTRA_VARS input to your seed job in addition
+    to those listed in the seed job.
+
+    APPS_TO_REPO: a dictionary containing mappings from an edX IDA to the name of its respective GitHub repository
+    APPS_TO_BRANCH: a dictionary containing mappings from an edX IDA to the name of its main GitHub branch
+    CONFIGURATION_REPO_URL: URL of the configuration GitHub repository
+    CONFIGURATION_BRANCH: the main GitHub branch of the configuration repository
+    EDX_REPO_ROOT: the URL of the edX GitHub organization
+    CONFIG_JSON_FILE_CREDENTIAL_ID: ID of the credentials set up in Jenkins to log into DockerHub
+
+    Expected credentials - these will normally be set up on the Folder.
+
+    docker-config-json: a file credential that gives each job in the folder permission to push to DockerHub
+*/
+
+package devops.jobs
+
+import static org.edx.jenkins.dsl.YAMLHelpers.parseYaml
+import org.yaml.snakeyaml.error.YAMLException
+
+class DockerCI {
+    public static def job = { dslFactory, extraVars ->
+        // for each application
+        extraVars.get("APPS_TO_REPO").each { app_name, app_repo ->
+            dslFactory.job('DockerCI/' + app_name) {
+
+                assert extraVars.containsKey('APPS_TO_REPO') : "Please define APPS_TO_REPO"
+                assert extraVars.containsKey('APPS_TO_BRANCH') : "Please define APPS_TO_BRANCH"
+                assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
+                assert extraVars.containsKey('CONFIGURATION_BRANCH') : "Please define CONFIGURATION_BRANCH"
+                assert extraVars.containsKey('EDX_REPO_ROOT') : "Please define EDX_REPO_ROOT"
+
+                // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
+                wrappers {
+                    credentialsBinding {
+                        file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                    }
+                }
+                multiscm {
+                    // check out edx/configuration repository from GitHub; necessary to be able to run dependency analyzer
+                    git {
+                        remote {
+                            url(extraVars.get("CONFIGURATION_REPO_URL"))
+                            branch(extraVars.get("CONFIGURATION_BRANCH"))
+                        }
+                        extensions {
+                            relativeTargetDirectory('configuration')
+                            cleanAfterCheckout()
+                            // ignore notifications on commits to branch; this job will be triggered by
+                            // configuration-watcher job
+                            ignoreNotifyCommit()
+                            cloneOptions {
+                                // increase clone timeout to 3 hours
+                                timeout(360)
```

Comment:
If this is minutes, it is 6 hours.  Was this a result of the old slow disk on the old docker image?  Do we need it?